gdx/src/com/badlogic/gdx/utils/JsonSkimmer.java

```diff
@@ -0,0 +1,592 @@
+		}
+		parse(data, 0, offset);
+	}
+
+	public void parse (InputStream input) {
+		Reader reader;
+		try {
+			reader = new InputStreamReader(input, "UTF-8");
+		} catch (Exception ex) {
+			throw new SerializationException("Error reading stream.", ex);
+		}
+		parse(reader);
+	}
+
+	public void parse (FileHandle file) {
+		Reader reader;
+		try {
+			reader = file.reader("UTF-8");
+		} catch (Exception ex) {
+			throw new SerializationException("Error reading file: " + file, ex);
+		}
+		try {
+			parse(reader);
+		} catch (Exception ex) {
+			throw new SerializationException("Error parsing file: " + file, ex);
+		}
+	}
+
+	public void parse (char[] data, int offset, int length) {
+		stop = false;
+		int cs, p = offset, pe = length, eof = pe, top = 0;
+		int[] stack = new int[4];
+
+		int s = 0;
+		String name = null;
+		boolean needsUnescape = false, stringIsName = false, stringIsUnquoted = false;
+		RuntimeException parseRuntimeEx = null;
+
+		boolean debug = false;
+		if (debug) System.out.println();
+
+		try {
+
+			// line 99 "../../../../../src/com/badlogic/gdx/utils/JsonSkimmer.java"
+			{
+				cs = json_start;
+				top = 0;
+			}
+
+			// line 104 "../../../../../src/com/badlogic/gdx/utils/JsonSkimmer.java"
+			{
+				int _klen;
+				int _trans = 0;
+				int _acts;
+				int _nacts;
+				int _keys;
+				int _goto_targ = 0;
+
+				_goto:
+				while (true) {
+					switch (_goto_targ) {
+					case 0:
+						if (p == pe) {
+							_goto_targ = 4;
+							continue _goto;
+						}
+						if (cs == 0) {
+							_goto_targ = 5;
+							continue _goto;
+						}
+					case 1:
+						_match:
+						do {
+							_keys = _json_key_offsets[cs];
+							_trans = _json_index_offsets[cs];
+							_klen = _json_single_lengths[cs];
+							if (_klen > 0) {
+								int _lower = _keys;
+								int _mid;
+								int _upper = _keys + _klen - 1;
+								while (true) {
+									if (_upper < _lower) break;
+
+									_mid = _lower + ((_upper - _lower) >> 1);
+									if (data[p] < _json_trans_keys[_mid])
+										_upper = _mid - 1;
+									else if (data[p] > _json_trans_keys[_mid])
+										_lower = _mid + 1;
+									else {
+										_trans += (_mid - _keys);
+										break _match;
+									}
+								}
+								_keys += _klen;
+								_trans += _klen;
+							}
+
+							_klen = _json_range_lengths[cs];
+							if (_klen > 0) {
+								int _lower = _keys;
+								int _mid;
+								int _upper = _keys + (_klen << 1) - 2;
+								while (true) {
+									if (_upper < _lower) break;
+
+									_mid = _lower + (((_upper - _lower) >> 1) & ~1);
+									if (data[p] < _json_trans_keys[_mid])
+										_upper = _mid - 2;
+									else if (data[p] > _json_trans_keys[_mid + 1])
+										_lower = _mid + 2;
+									else {
+										_trans += ((_mid - _keys) >> 1);
+										break _match;
+									}
+								}
+								_trans += _klen;
+							}
+						} while (false);
+
+						_trans = _json_indicies[_trans];
```

Comment:
Minor typo: `indicies` should be `indices` . This is only used internally, so it doesn't really matter.

Reply:
Yeah that looks odd. It's generated though.